tests/functional/io_apis_test.c

```diff
@@ -0,0 +1,832 @@
+
+static void
+load_thread_params(io_test_fn_args *io_test_param,
+                   io_test_fn_args *thread_params,
+                   int              nthreads);
+
+static platform_status
+do_n_thread_creates(const char         *thread_type,
+                    uint64              num_threads,
+                    io_test_fn_args    *params,
+                    test_io_thread_hdlr thread_hdlr);
+
+void
+test_sync_writes_worker(void *arg);
+
+void
+test_sync_reads_worker(void *arg);
+
+void
+test_async_reads_worker(void *arg);
+
+/*
+ * Helper inline functions.
+ */
+/*
+ * Convert the range {start, end}-address of the device on which we
+ * want to do IO to # of pages. Distribute this across # of threads to
+ * return the # of pages / thread to perform IO.
+ */
+static inline uint64
+npages_per_thread(io_test_fn_args *io_test_param, int nthreads)
+{
+   return (((io_test_param->end_addr - io_test_param->start_addr)
+            / io_test_param->io_cfgp->page_size)
+           / nthreads);
+}
+
+/*
+ * ----------------------------------------------------------------------------
+ * splinter_io_apis_test() - Entry point 'main' for SplinterDB IO APIs testing.
+ * ----------------------------------------------------------------------------
+ */
+int
+splinter_io_apis_test(int argc, char *argv[])
+{
+   uint64 heap_capacity = (256 * MiB); // small heap is sufficient.
+
+   // Create a heap for io system's memory allocation.
+   platform_heap_handle hh  = NULL;
+   platform_heap_id     hid = NULL;
+   platform_status      rc =
+      platform_heap_create(platform_get_module_id(), heap_capacity, &hh, &hid);
+   platform_assert_status_ok(rc);
+
+   // Do minimal IO config setup, using default IO values.
+   master_config master_cfg;
+
+   // Initialize the IO sub-system configuration.
+   ZERO_STRUCT(master_cfg);
+   config_set_defaults(&master_cfg);
+
+   // Parse config-related command-line arguments. Only expecting to support:
+   // --verbose-progress
+   rc = config_parse(&master_cfg, 1, (argc - 1), (argv + 1));
+   if (!SUCCESS(rc)) {
+      return -1;
+   }
+
+   Verbose_progress = master_cfg.verbose_progress;
+
+   // Ensure that the default max async q-depth configured for
+   // master-cfg is sufficiently big enough for this test case.
+   platform_assert(NUM_PAGES_RW_ASYNC_PER_THREAD
+                      <= master_cfg.io_async_queue_depth,
+                   "NUM_PAGES_RW_ASYNC_PER_THREAD=%d, "
+                   "Master io_async_queue_depth=%lu\n",
+                   NUM_PAGES_RW_ASYNC_PER_THREAD,
+                   master_cfg.io_async_queue_depth);
+
+   io_config io_cfg;
+   ZERO_STRUCT(io_cfg);
+
+   io_config_init(&io_cfg,
+                  master_cfg.page_size,
+                  master_cfg.extent_size,
+                  master_cfg.io_flags,
+                  master_cfg.io_perms,
+                  master_cfg.io_async_queue_depth,
+                  "splinter_io_apis_test.db");
```

Comment:
Could you choose a name that matches this glob?
https://github.com/vmware/splinterdb/blob/83de5e03722d61436ce247e406d87b8fa4913d85/.gitignore#L11

That way my git working tree stays clean when running this test.

Reply:
Done.

Reply:
@gapisback I'm sorry, but it still isn't fixed even with https://github.com/vmware/splinterdb/pull/473/commits/16738a6e21642d96c7b815f14b20f78c9140514f 

You'll need to have the filename start with `splinterdb` note the "db" at the end.

Alternately, change the `.gitignore` file to cover this file.

Reply:
Oooops, sorry, @rosenhouse - I didn't quite get it the last time. Didn't read the `r.e` very carefully. Just pushed a commit to rename this to `splinterdb_io_apis_test_db`, which should sort out the problem.

Reply:
Yup, fixed now, thanks!